instance.tf

```diff
@@ -0,0 +1,30 @@
+resource "aws_instance" "web_server" {
+  ami                         = "ami-051dfed8f67f095f5"
+  associate_public_ip_address = true
+  instance_type               = "t3.micro"
+  subnet_id                   = aws_subnet.public[0].id
+  key_name                    = "ec2key"
+  vpc_security_group_ids      = [aws_security_group.allow_ssh.id]
+}
+
+resource "aws_security_group" "allow_ssh" {
+  name        = "allow_ssh"
+  description = "Allow SSH inbound traffic"
+  vpc_id      = aws_vpc.main.id
+
+  ingress {
+    from_port        = 22
+    to_port          = 22
+    protocol         = "tcp"
+    cidr_blocks      = ["0.0.0.0/0"]
```

Comment:
`0.0.0.0/0` means everyone from the internet can connect. I suggest that you limit this rule to your public IP only. 
You can find out your public IP using this website https://whatismyipaddress.com/